### ISSUE TITLE
Add how to gather single piece of text data

### ISSUE THREAD

**Issue by kelliewellingtons** [barnardos/design-system#65]
_As a user of the design system
I want to know how to add a single text input component to my project
So that there is a consistent experience across services_

## Description

Submitting data is a fundamental aspect of interfaces.  The most common form of data is freeform text. There are a number of problems surrounding input:

![img_20180308_151535](https://user-images.githubusercontent.com/808227/37164153-71df7b26-22f1-11e8-9227-bec019342882.jpg)

## Links and resources

- https://govuk-elements.herokuapp.com/form-elements/
- https://paper.dropbox.com/doc/Text-boxes-8NLlgz9tjR8OzTz2N9wE3
- https://paper.dropbox.com/doc/Radio-buttons-oIwWoxwBKClt5IXvDbnpF
- https://paper.dropbox.com/doc/Checkboxes-HDLVh9YAhInay2xJ7vBl8
- https://paper.dropbox.com/doc/Error-messages-fifEJpOYMGjRy0lHTmthb
- https://webaim.org/techniques/forms/

Website:

![img_20180308_151110](https://user-images.githubusercontent.com/808227/37163624-28d79f0e-22f0-11e8-8cce-3fcf0a93ca57.jpg)

Content form:

<img width="884" alt="screen shot 2018-03-08 at 15 51 26" src="https://user-images.githubusercontent.com/808227/37163666-3d3e5348-22f0-11e8-918a-3c3fb77a1470.png">

## Acceptance Criteria

Document solutions to following concerns:

- [x] Consistent style
- [x] Size of the component relative to expect the size of the input
- [x] Label/micro copy
- [x] Label and input relationship (position etc)
- [x] Placeholder/example/hint text
- [x] Focus state
- [x] Mandatory vs optional
- [x] Single line vs Multi-line
- [x] Invalid/error states
- [x] Question why/whether the data should be captured

Then:

- [x] Add component to design system
- [x] Addresses accessibility/usability considerations
- [x] Works on target devices and browsers

**Comment by kelliewellingtons** [comment]
Key findings from [gov.uk](https://govuk-elements.herokuapp.com/form-elements/)

> Keep forms as simple as possible

> Optional and mandatory fields
> - only ask for the information you absolutely need
> - if you do ask for optional information, mark the labels of optional fields with ‘(optional)’
> - don’t mark mandatory fields with asterisks

**Comment by jeddy3** [comment]
[NN Group findings](https://www.nngroup.com/articles/web-form-design/):

> - Keep it short - eliminate unnecessary fields
> - Visually group related labels and fields
> - Present fields in a single column layout. 
> - Use logical sequencing.
> - [Avoid placeholder text](https://www.nngroup.com/articles/form-design-placeholders/).
> - Match fields to the type and size of the input. 
> - Distinguish optional and required fields
> - Explain any input or formatting requirements.
> - Avoid Reset and Clear buttons. 
> - Provide highly visible and specific error messages.

**Comment by jeddy3** [comment]
On naming... the [W3C Web Accessibility uses the term "control"](https://www.w3.org/WAI/tutorials/forms/grouping/) to group labels, inputs and associated meta:

> Grouping related form controls makes forms more understandable for all users, as related controls are easier to identify. It also makes it easier for people to focus on smaller and more manageable groups rather than try to grasp the entire form at once.

> Grouping needs to be carried out visually and in the code, for example, by using the `<fieldset>` and `<legend>` elements to associate related form controls.

**Comment by kelliewellingtons** [comment]
Key findings from [Co-op Design System](http://coop-design-manual.herokuapp.com/styles/forms/index.html)

> - Know why you’re asking each question and what it’s needed for
> - Tell users why we’re asking
> - If the answer relies on the users' understanding of a complex concept, explain it in plain English. Do not assume any prior knowledge of the subject you’re writing about, or a certain level of literacy, education or digital experience of your users. Design for everyone
> allow users to click on the question or label to let them start answering (doing this takes the cursor to the text entry field)
> - do not use placeholder text (text that sits within the text entry field) — if a question needs extra clarification, use hint text instead (static text that sits outside of the text entry field)
> - make the length of the text input box appropriate for the amount of information needed: do not make it frustratingly short or intimidatingly long.

**Comment by kelliewellingtons** [comment]
**Key findings relating to marking fields as optional as opposed to mandatory**

[UX Movement
](http://uxmovement.com/forms/why-users-fill-out-less-if-you-mark-required-fields/) Why Users Fill Out Less If You Mark Required Fields

> - Marking optional fields won't inhibit user's voluntary over-disclosure

[Optimal Workshop](https://blog.optimalworkshop.com/designing-form-fields-optional-versus-required) Designing form fields: Optional versus required

> - Including the phrase ‘optional’ after a label is much clearer than any visual symbol you could use to mean the same thing. 
> - Using language like “Optional” is letting users know they have a choice, which is inline with the Norman Nielsen Group usability principle of user control and freedom.
> - The asterisk is a design element that is cluttering up a form. It’s taking away from the goal of filling out the form. Chances are the majority of your form fields will be required so only needing to mark the optional ones takes up less real estate.
> - It’s expected that fields are required.
> - Only ask what is necessary

[Fusion Box
](https://www.fusionbox.com/blog/detail/rethinking-the-red-required-asterisk-for-better-form-ux/599/) Rethinking The Red "Required" Asterisk For Better Form UX

> - While the asterisk symbol is a long-held convention, it remains a potentially ambiguous icon without a clear semantic meaning.
> - This clutter also affects visually impaired users and those relying on assistive technologies such as screen readers. Depending on the software package and user settings asterisks are either ignored or read out loud as "star". Appropriate ARIA properties will ensure that required fields are clearly delineated, but adding these labels will not guarantee that the asterisk will be skipped over.
> - When approaching forms in which a majority of the fields are mandatory it is often best to mark only the optional fields and allow users to infer that unmarked fields are required.

**Comment by kelliedesigner** [comment]
[**Design Better Forms**](https://uxdesign.cc/design-better-forms-96fadca0f49c)
Common mistakes designers make and how to fix them

**Comment by kelliedesigner** [comment]
[Placeholders in Form Fields Are Harmful
](https://www.nngroup.com/articles/form-design-placeholders/)
